kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -207,11 +209,10 @@ public ConfigurationServiceImpl() {
     //
     // ----------------------------------------------------------------
 
-    protected void activate(ComponentContext componentContext) throws InvalidSyntaxException {
+    protected void activate() throws InvalidSyntaxException {
         s_logger.info("activate...");
 
-        // save the bundle context
-        this.m_ctx = componentContext;
+        this.bundleContext = FrameworkUtil.getBundle(ConfigurationServiceImpl.class).getBundleContext();
```

Comment:
Yes, we can do this. We also don't need to pass in the ComponentContext as the DS layer will figure out the right parameters for the method anyway.